### PR TITLE
Added "Target" input

### PR DESCRIPTION
**Pull request by Supertos**
Works simular to gmod_wire_gimbal. Makes ranger do traces in provided world position direction

### CODE REVIEW

**Comment by Supertos** [comment]
Fixed static beam

**Comment by thegrb93** [comment]
I think we prefer ~= instead of !=

**Comment by Supertos** [comment]
> I think we prefer ~= instead of !=

I'm unsure of who's original wiremod programmer but they do love using non-vanila syntax

**Comment by Supertos** [comment]
> I think we prefer ~= instead of !=

I'm unsure of who's original wiremod programmer but they do love using non-vanila syntax

File: lua/entities/gmod_wire_ranger.lua
Comment (thegrb93) [inline lua/entities/gmod_wire_ranger.lua:125]: Could you replace Vector() with vector_origin

File: lua/entities/gmod_wire_ranger.lua
Comment (thegrb93) [inline lua/entities/gmod_wire_ranger.lua:126]: Change `:Angle():Forward()` to `:GetNormalized()` and on the line after it, `if tracedata.endpos[1] ~= tracedata.endpos[1] then tracedata.endpos = Vector(self:GetBeamLength(), 0, 0) end`

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:167]: replace :Angle():Forward() here too.

**Review by thegrb93 — CHANGES_REQUESTED** [review]
Should be good to merge after these optimizations

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:196]: Does the 'end' above this else need to be removed?

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:167]: Need the `ent.endpos[1] ~= ent.endpos[1]` check here too

Comment (TomyLobo) [reply lua/wire/client/cl_wirelib.lua:167]: Why would you check if something is different from itself?

File: lua/entities/gmod_wire_ranger.lua
Comment (TomyLobo) [inline lua/entities/gmod_wire_ranger.lua:127]: This check looks like it can never be true, cause you're checking if something is unequal to itself.

Comment (thegrb93) [reply lua/entities/gmod_wire_ranger.lua:127]: It's true if the value is nan

File: lua/wire/client/cl_wirelib.lua
Comment (Divran) [reply lua/wire/client/cl_wirelib.lua:167]: We also have https://github.com/wiremod/wire/blob/master/lua/wire/wireshared.lua#L949-L951 if you'd like to use that

**Comment by thegrb93** [comment]
The 'end' I mentioned above still needs to be addressed

**Comment by TomyLobo** [comment]
as does the strange condition that's always false.
@thegrb93 can you keep track of that? I'm not sure I'll get to see the code again before you merge it.

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:166]: These if cases should in included in line 165, otherwise the Skew inputs block won't work.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:197]: Moving this down here causes these vars to be undefined in the above code. Can you also rename 'start' to 'pos' and replace all occurrences of ent:GetPos() with pos?

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:79]: Please don't align assignments using spaces

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:56]: I don't agree this should be included in this PR, adding it could cause more issues and would be hard to pinpoint

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:204]: The old code was correct, it draws two beams if hilight. This code only draws one beam.

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:188]: These are needed

**Review by thegrb93 — CHANGES_REQUESTED** [review]
More issues now

**Comment by Supertos** [comment]
> Please don't align assignments

@thegrb93 can you tell me why i shouldn't align assignments?

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [reply lua/wire/client/cl_wirelib.lua]: Ok nevermind, I thought you aded this but you actually moved it. Might be fine.

**Comment by thegrb93** [comment]
> > Please don't align assignments
> 
> @thegrb93 can you tell me why i shouldn't align assignments?

It creates more work for whoever wants to modify those code sections or add new variables. Adding a new variable can lead to needing to realign every other var. It's just all around more annoying.

**Comment by thegrb93** [comment]
I think I just have to test and check it again and it'll be ok to merge

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [inline lua/wire/client/cl_wirelib.lua:55]: This will break the beam rendering if there are no wires to render.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/wire/client/cl_wirelib.lua
Comment (thegrb93) [reply lua/wire/client/cl_wirelib.lua:55]: Nvm that's a different function

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**
